src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1048,6 +1048,7 @@ private void AddTransactionToWallet(Transaction transaction, TxOut utxo, int? bl
                     IsCoinStake = transaction.IsCoinStake == false ? (bool?)null : true,
                     BlockHeight = blockHeight,
                     BlockHash = block?.GetHash(),
+                    BlockIndex = block?.Transactions.FindIndex(t => t.GetHash() == transactionHash),
```

Comment:
Are we OK with this being `-1` or would `null` be preferable? I don't mind either way if it's consistent with other APIs.

Reply:
You mean if findIndex doesn't find a match?
It has to, we know for a fact that the block contains the transaction.